atlascope/core/rest/endpoints/tile_endpoints.py

```diff
@@ -1,135 +1,72 @@
 
 
-class TileMetadataView(GenericAPIView, mixins.RetrieveModelMixin):
+class DatasetTileSourceView(GenericViewSet, LargeImageDetailMixin):
     queryset = Dataset.objects.filter(content__isnull=False, dataset_type='tile_source')
-    serializer_class = TileMetadataSerializer
+    serializer_class = DatasetSerializer
 
-    def get(self, *args, **kwargs):
+    default_colors = ['#ffffff']
+
+    def get_path(self, *args, **kwargs) -> str:
+        """Retreive path to image file from data record."""
         dataset = self.get_object()
-        content_location = dataset.content.path
         try:
-            tile_source = OMETiffFileTileSource(content_location)
-        except TileSourceError:
-            tile_source = TiffFileTileSource(content_location)
-        serializer = self.get_serializer(tile_source)
-        return Response(serializer.data)
+            return str(dataset.content.path)
+        except (AttributeError, ValueError):
+            # Raise 400-level error as this dataset has no content
+            raise ValidationError('Dataset has not content.')
 
+    def open_tile_source(self, path: str, *args, **kwargs) -> FileTileSource:
+        """Override to manually choose tile source class."""
+        try:
+            tile_source = OMETiffFileTileSource(path, *args, **kwargs)
+        except TileSourceError:
+            tile_source = TiffFileTileSource(path, *args, **kwargs)
+        return tile_source
 
-class LargeImageRenderer(BaseRenderer):
-    media_type = 'image/png'
-    format = 'png'
-
-    def render(self, data, media_type=None, renderer_context=None):
-        return data
+    def get_style(self, request: Request) -> dict:
+        """Override django-large-image style parsing for cutom frame stuff.
 
+        This builds a style dictionary for large-image following:
 
-class TileView(GenericAPIView, mixins.RetrieveModelMixin):
-    queryset = Dataset.objects.filter(content__isnull=False, dataset_type='tile_source')
-    model = Dataset
-    renderer_classes = [LargeImageRenderer]
+            https://girder.github.io/large_image/tilesource_options.html#style
 
-    default_colors = ['#ffffff']
+        """
+        channels = request.query_params.get('channels')
+        if channels:
+            channels = channels.split(',')
+        else:
+            tile_source = self.open_tile_source(self.get_path())  # TODO: better handle upstream
+            channels = range(len(tile_source.getMetadata()['frames']))
+        colors = request.query_params.get('colors')
+        if colors:
+            colors = [f'#{color}' for color in colors.split(',')]
+        else:
+            colors = self.default_colors
+        style = {'bands': []}
+        for channel, color in list(zip(channels, cycle(colors))):
+            style['bands'].append(
+                {
+                    'frame': channel,
+                    'palette': ['#000', color],
+                }
+            )
+        return style
```

Comment:
Ideally, we should not override this method, but instead, have the front-end create the JSON style per https://github.com/girder/django-large-image#-styling and https://girder.github.io/large_image/tilesource_options.html#style

Reply:
This would also prevent opening the source twice - note that it isn't cached because the first open is without a style and the second is with a style

Reply:
@naglepuff Do you think you'd be able to accomplish this frontend refactoring? If so, it might be appropriate to make another PR that goes into this one

Reply:
Merging this and having a follow up front-end PR seems appropriate to me

Reply:
Yeah, for sure I can take that on